src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1635,6 +1639,40 @@ else if (null == currentPrimaryPlaceHolder) {
                     currentConnectPlaceHolder = currentPrimaryPlaceHolder;
                 }
 
+                try {
+                    InetAddress[] inetAddrs = InetAddress.getAllByName(currentConnectPlaceHolder.getServerName());
+
+                    // if only one IP is resolved and user didn't set TNIR explicitly then set TNIR to false
+                    if ((1 == inetAddrs.length) && !userSetTNIR) {
```

Comment:
@v-xiangs I made a comment about DNS resolution for TNIR to be removed here https://github.com/Microsoft/mssql-jdbc/pull/209#issuecomment-291694425 
The problem is that Single IP address could be possible in case of a Single Subnet where Timeslicing may be needed for AGs